src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelector.java

```diff
@@ -80,6 +80,24 @@ private static boolean matches(@Nonnull Class<?>[] parameterTypes, @Nonnull Obje
         return true;
     }
 
+    private static boolean isInstancePrimitive(@Nonnull Class<?> type, @Nonnull Object instance) {
+        if (type.isInstance(instance)) {
+            return true;
+        }
+        // https://docs.oracle.com/javase/specs/jls/se8/html/jls-5.html#jls-5.1.2
+        if (instance instanceof Number) {
+            Number n = (Number) instance;
+            if (type == Long.class || type == Double.class) {
+                return true;
```

Comment:
Yes, the conversions between numeric types leave a lot of possibilities.

I just throw in `double` here although it is rarely used in interesting API methods. What I care about, and tested, is a method declared to take `long` but being passed an `int`, which caused @cyrille-leclerc pain. I am not interested in spending a lot of time fully implementing the JLS definitions unless there are known use cases, or there is some library out there which does the heavy lifting.

What we really want is for the Groovy runtime to select the actual `Method` (etc.) it is about to call, then pass that information to the `GroovyInterceptor` right before the call, so that we do not need to be endlessly tuning this plugin for special cases (in fact this whole class would disappear). Unfortunately the sandbox does not do that yet. See https://github.com/kohsuke/groovy-sandbox/issues/7.